tests/mock_tests/sfloworh_ut.cpp

```diff
@@ -0,0 +1,269 @@
+            // PortsOrch
+            //
+
+            const int portsorchBasePri = 40;
+
+            std::vector<table_name_with_pri_t> portTableList = {
+                { APP_PORT_TABLE_NAME,        portsorchBasePri + 5 },
+                { APP_VLAN_TABLE_NAME,        portsorchBasePri + 2 },
+                { APP_VLAN_MEMBER_TABLE_NAME, portsorchBasePri     },
+                { APP_LAG_TABLE_NAME,         portsorchBasePri + 4 },
+                { APP_LAG_MEMBER_TABLE_NAME,  portsorchBasePri     }
+            };
+
+            gPortsOrch = new PortsOrch(this->appDb.get(), this->stateDb.get(), portTableList, this->chassisAppDb.get());
+            gDirectory.set(gPortsOrch);
+            resourcesList.push_back(gPortsOrch);
+
+            //
+            // FlexCounterOrch
+            //
+            std::vector<std::string> flexCounterTableList = {
+                CFG_FLEX_COUNTER_TABLE_NAME
+            };
+
+            auto flexCounterOrch = new FlexCounterOrch(this->configDb.get(), flexCounterTableList);
+            gDirectory.set(flexCounterOrch);
+            resourcesList.push_back(flexCounterOrch);
+        }
+
+        void deinitOrch()
+        {
+            std::reverse(this->resourcesList.begin(), this->resourcesList.end());
+            for (auto &it : this->resourcesList)
+            {
+                delete it;
+            }
+
+            gPortsOrch = nullptr;
+            gDirectory.m_values.clear();
+        }
+
+        void initPorts()
+        {
+            auto portTable = Table(this->appDb.get(), APP_PORT_TABLE_NAME);
+
+            // Get SAI default ports to populate DB
+            auto ports = ut_helper::getInitialSaiPorts();
+
+            // Populate port table with SAI ports
+            for (const auto &cit : ports)
+            {
+                portTable.set(cit.first, cit.second);
+            }
+
+            // Set PortConfigDone
+            portTable.set("PortConfigDone", { { "count", to_string(ports.size()) } });
+            gPortsOrch->addExistingData(&portTable);
+            static_cast<Orch*>(gPortsOrch)->doTask();
+
+            // Set PortInitDone
+            portTable.set("PortInitDone", { { "lanes", "0" } });
+            gPortsOrch->addExistingData(&portTable);
+            static_cast<Orch*>(gPortsOrch)->doTask();
+        }
+
+        void initDb()
+        {
+            this->appDb = std::make_shared<swss::DBConnector>("APPL_DB", 0);
+            this->configDb = std::make_shared<swss::DBConnector>("CONFIG_DB", 0);
+            this->stateDb = std::make_shared<swss::DBConnector>("STATE_DB", 0);
+            this->chassisAppDb = std::make_shared<swss::DBConnector>("CHASSIS_APP_DB", 0);
+        }
+
+        std::shared_ptr<DBConnector> appDb;
+        std::shared_ptr<DBConnector> configDb;
+        std::shared_ptr<DBConnector> stateDb;
+        std::shared_ptr<DBConnector> chassisAppDb;
+
+        std::vector<Orch*> resourcesList;
+    };
+
+    TEST_F(SflowOrchTest, test)
```

Comment:
please consider having a different tests for different SFlow objects (e.g., global, session, etc.)